### DISCUSSION TITLE
For performance reasons, is disabling event listeners necessary when using the usePopper hook?

### DISCUSSION THREAD

**Discussion by jtabone16** [floating-ui/floating-ui#1385] [Q&A]
The[ tutorial part of the docs has a performance section](https://popper.js.org/docs/v2/tutorial/#performance) that mentions disabling event listeners when the tooltip is hidden for performance reasons. Is this needed when using the usePopper hook since the hook[ doesn't return a setOptions function](https://popper.js.org/react-popper/v2/hook/#update-forceupdate-and-state)?

**Comment by FezVrasta** [comment]
react-popper is declarative, you can pass a new set of options to the hook and they will be applied

**Reply by jtabone16** [reply]
ok thought so! thanks for the quick reply
